lsp-methods.el

```diff
@@ -1894,14 +1859,35 @@ It will be used when no language has been specified in document/onHover result."
   (cl-check-type renderer function)
   (setf (lsp--client-default-renderer client) renderer))
 
+(defvar-local lsp--hover-saved-bounds nil)
+(defvar-local lsp--hover-saved-contents nil)
+
+(defun lsp--hover-callback (from-cache)
+  (message lsp--hover-saved-contents))
+
 (defun lsp-hover ()
   "Show relevant documentation for the thing under point."
   (interactive)
-  (lsp--text-document-hover-string))
+  (if (and lsp--hover-saved-bounds
+           (lsp--point-in-bounds-p lsp--hover-saved-bounds))
+      (lsp--hover-callback t)
+    (lsp--send-request-async
+     (lsp--make-request "textDocument/hover"
+                        (lsp--text-document-position-params))
+     (lambda (hover)
```

Comment:
you may write it like this:
``` emacs-lisp
(-lambda ((&hash "contents" "range")) ...)
```

Reply:
Changed to (-lambda ((&hash "contents" "range")) ...)

I have to update both `lsp.el` and `lsp-methods.el` because I can update `lsp-ui-doc--hover` to advice `lsp--hover-callback` as a follow-up without breaking both interfaces.